ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -340,10 +341,14 @@ private static void forwardTo(
         .map(location -> Integer.toString(location.getPartition()))
         .collect(Collectors.joining(","));
     // Add skip forward flag to properties
-    final Map<String, Object> requestProperties = ImmutableMap.of(
-        KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_SKIP_FORWARDING, true,
-        KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST, true,
-        KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_PARTITIONS, partitions);
+    final Map<String, Object> requestProperties = new HashMap<>();
+    requestProperties.put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_SKIP_FORWARDING, true);
+    requestProperties.put(KsqlRequestConfig.KSQL_REQUEST_INTERNAL_REQUEST, true);
+    requestProperties.put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_PARTITIONS, partitions);
+    if (consistencyOffsetVector.isPresent()) {
+      requestProperties.put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR,
```

Comment:
I think we also have to add the "consistency vector enabled" config, right, @vpapavas ?

Which raises the question: is this logic already being tested by the existing tests from your prior PR? (If so, and we are missing the other config, then how is that test passing?)

Reply:
On the server side we check that:

1. The server config KsqlConfig.KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED is set to true
2. The request properties contain a consistency token (may be an empty string the first time a request is made)

In the tests, @patrickstuedi has set the server property to true hence only the offset needs to be added.

We don't additionally need to check if the client has set the "consistency vector enabled" config, since the only way to add the vector to the request properties, is if this config is set. In this case here, the Optional will be non-empty, only if consistency is enabled